exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/NetworkFunctions.java

```diff
@@ -0,0 +1,668 @@
+    }
+
+    public void eval() {
+
+      String cidrString = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(inputCIDR.start, inputCIDR.end, inputCIDR.buffer);
+      utils = new org.apache.commons.net.util.SubnetUtils(cidrString);
+
+      String outputValue = utils.getInfo().getHighAddress();
+
+      out.buffer = buffer;
+      out.start = 0;
+      out.end = outputValue.getBytes().length;
+      buffer.setBytes(0, outputValue.getBytes());
+
+    }
+  }
+
+  /**
+   * This function encodes URL strings.
+   */
+  @FunctionTemplate(
+    name = "urlencode",
+    scope = FunctionTemplate.FunctionScope.SIMPLE,
+    nulls = FunctionTemplate.NullHandling.NULL_IF_NULL
+  )
+  public static class urlencodeFunction implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder inputString;
+
+    @Output
+    VarCharHolder outputString;
+
+    @Inject
+    DrillBuf buffer;
+
+    public void setup() {
+    }
+
+    public void eval() {
+
+      String url = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(inputString.start, inputString.end, inputString.buffer);
+
+      String outputValue = "";
+      try {
+        outputValue = java.net.URLEncoder.encode(url, "UTF-8");
+      }catch (Exception e){
+
+      }
+      outputString.buffer = buffer;
+      outputString.start = 0;
+      outputString.end = outputValue.getBytes().length;
+      buffer.setBytes(0, outputValue.getBytes());
+
+    }
+  }
+
+  /**
+   * This function decodes URL strings.
+   */
+  @FunctionTemplate(
+    name = "urldecode",
+    scope = FunctionTemplate.FunctionScope.SIMPLE,
+    nulls = FunctionTemplate.NullHandling.NULL_IF_NULL
+  )
+  public static class urldecodeFunction implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder inputString;
+
+    @Output
+    VarCharHolder outputString;
+
+    @Inject
+    DrillBuf buffer;
+
+    public void setup() {
+    }
+
+    public void eval() {
+
+      String url = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(inputString.start, inputString.end, inputString.buffer);
+
+      String outputValue = "";
+      try {
+        outputValue = java.net.URLDecoder.decode(url, "UTF-8");
+      }catch (Exception e){
+
+      }
+      outputString.buffer = buffer;
+      outputString.start = 0;
+      outputString.end = outputValue.getBytes().length;
+      buffer.setBytes(0, outputValue.getBytes());
+
+    }
+  }
+
+
+  /**
+   * This function converts a BigInt IPv4 into dotted decimal notation.  The opposite of inet_aton.
+   */
+
+  @FunctionTemplate(
+    name = "inet_ntoa",
+    scope = FunctionTemplate.FunctionScope.SIMPLE,
+    nulls = FunctionTemplate.NullHandling.NULL_IF_NULL
+  )
+  public static class InetNtoaFunction implements DrillSimpleFunc {
+
+    @Param
+    BigIntHolder in1;
+
+    @Output
+    VarCharHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+
+    public void setup() {
+    }
+
+
+    public void eval() {
```

Comment:
For `ntoa` and `aton` may we use guava lib [1, 2]?

[1] https://google.github.io/guava/releases/19.0/api/docs/com/google/common/net/InetAddresses.html
[2] https://stackoverflow.com/questions/2241229/going-from-127-0-0-1-to-2130706433-and-back-again

Reply:
Is there a preference to using an existing library?

Reply:
Well, it depends but I would call it's Java way to reuse already defined logic from different libraries rather then repeating it. Of course if it's worse adding new lib. In our case logic for `ntoa` , `aton` and `private_ip` does not seem trivial. Did you write it yourself or copied from somewhere else? If first case according to which references algorithm was written? Were all cases considered, tested etc? If second, is source is well-trusted? Does source owner mind his code will be reused etc? Maybe it's better to use well-known lib instead which mostly likely has all these functions tested and followed common standards?